tests/back/Integration/IntegrationTestsBundle/Messenger/AssertEventCountTrait.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Test\IntegrationTestsBundle\Messenger;
+
+use Akeneo\Platform\Component\EventQueue\BulkEventInterface;
+use Akeneo\Platform\Component\EventQueue\Event;
+use Symfony\Component\Messenger\TraceableMessageBus;
+
+trait AssertEventCountTrait
+{
+    public function assertEventCount(int $expectedCount, string $eventClassName): void
```

Comment:
```suggestion
    public function assertEventCount(int $expectedCount, string $eventName): void
```
I'm not sure we should directly use the Events class here, to avoid being "linked" to the connectivity context.
Using the type directly, like "product.created" we would at least be decoupled properly.

or create an assert for each type of events
```
assertProductCreatedEventCount(int $expectedCount)
```

Reply:
Following a discussion with @willy-ahva . I take back this suggestion.
The code here is clear enough and there is probably no issue with the bounded context / coupling